esp32_ulp/opcodes.py

```diff
@@ -267,6 +268,22 @@ def make_ins(layout):
 ARG = namedtuple('ARG', ('type', 'value', 'raw'))
 
 
+def eval_arg(arg):
+    parts = []
+    for token in split_tokens(arg):
+        if symbols.has_sym(token):
+            _, _, sym_addr = symbols.get_sym(token)
+            parts.append(str(sym_addr))
+        else:
+            parts.append(token)
+    parts = "".join(parts)
+    try:
+        parts = str(eval(parts))
+    except:
+        pass
```

Comment:
aka "the diaper pattern" :-)

Reply:
Continuing the pattern that was already there :). I always try to stick with styles/approaches used, rather than to impose my approach, because a file with many different styles mixed is worse in my opinion.

What I do like (a pattern, which I thought I was continuing), is the early-return-fall-through approach, where the last line of the function is the "if all else failed, do this". In case of the try-except-else you proposed further down, there is no last line on it's own but it's the else of a bigger structure. That potentially makes it more likely that one would have to untangle the try-except-else when making future changes (e.g. adding one more case) and require more care to make sure there remains an "if all else failed, do this" case (of course tests should catch this). For that reason, I would choose to keep it as is, but happy to go either way.

Reply:
can be done, if it can't be avoided.

about the try/except/else: the point was not only to move the operation into the else, but also to reduce the scope of the try-block to what really should be covered.

Reply:
Oh, sorry. I had not noticed that these comments related to two different parts of the code altogether. Somehow Github doesn't show enough context, so it looks like opening the file from the heading of the snippet being shown here, gives a better view. I will use that from now on.

You are very right. Catching anything and throwing it away is not the way to go here. I initially wrapped it that way, because inside eval anything can happen, so I didnt know what exactly to catch, but while adding the validation code yesterday, I actually noticed the try/except around the eval isnt even needed. When we get to that eval, it must evaluate to something or the expression is invalid. So in the latest commit (which I will rebase+squash) I removed this altogether.